src/d2d1.rs

```diff
@@ -237,40 +201,30 @@ STRUCT!{struct D2D1_LAYER_PARAMETERS {
     opacityBrush: *mut ID2D1Brush,
     layerOptions: D2D1_LAYER_OPTIONS,
 }}
-#[repr(i32)] #[derive(Copy, Clone, Debug)] #[allow(unused_qualifications)]
-pub enum D2D1_WINDOW_STATE {
+ENUM!{enum D2D1_WINDOW_STATE {
     D2D1_WINDOW_STATE_NONE = 0x0000000,
     D2D1_WINDOW_STATE_OCCLUDED = 0x0000001,
-}
-pub use self::D2D1_WINDOW_STATE::*;
-#[repr(i32)] #[derive(Copy, Clone, Debug)] #[allow(unused_qualifications)]
-pub enum D2D1_RENDER_TARGET_TYPE {
+}}
+ENUM!{enum D2D1_RENDER_TARGET_TYPE {
     D2D1_RENDER_TARGET_TYPE_DEFAULT = 0,
     D2D1_RENDER_TARGET_TYPE_SOFTWARE = 1,
     D2D1_RENDER_TARGET_TYPE_HARDWARE = 2,
-}
-pub use self::D2D1_RENDER_TARGET_TYPE::*;
-#[repr(i32)] #[derive(Copy, Clone, Debug)] #[allow(unused_qualifications)]
-pub enum D2D1_FEATURE_LEVEL {
+}}
+ENUM!{enum D2D1_FEATURE_LEVEL {
     D2D1_FEATURE_LEVEL_DEFAULT = 0,
-    D2D1_FEATURE_LEVEL_9 = ::D3D_FEATURE_LEVEL_9_1 as i32,
-    D2D1_FEATURE_LEVEL_10 = ::D3D_FEATURE_LEVEL_10_0 as i32,
-}
-pub use self::D2D1_FEATURE_LEVEL::*;
-#[repr(i32)] #[derive(Copy, Clone, Debug)] #[allow(unused_qualifications)]
-pub enum D2D1_RENDER_TARGET_USAGE {
+    D2D1_FEATURE_LEVEL_9 = ::D3D_FEATURE_LEVEL_9_1.0,
+    D2D1_FEATURE_LEVEL_10 = ::D3D_FEATURE_LEVEL_10_0.0,
+}}
+ENUM!{enum D2D1_RENDER_TARGET_USAGE {
     D2D1_RENDER_TARGET_USAGE_NONE = 0x00000000,
     D2D1_RENDER_TARGET_USAGE_FORCE_BITMAP_REMOTING = 0x00000001,
     D2D1_RENDER_TARGET_USAGE_GDI_COMPATIBLE = 0x00000002,
-}
-pub use self::D2D1_RENDER_TARGET_USAGE::*;
-#[repr(i32)] #[derive(Copy, Clone, Debug)] #[allow(unused_qualifications)]
-pub enum D2D1_PRESENT_OPTIONS {
+}}
+ENUM!{enum D2D1_PRESENT_OPTIONS {
```

Comment:
Also `FLAGS!`.
A whole bunch of these enums here that have hexadecimal discriminants should be `FLAGS!`.